src/components/main-view/main-view.jsx

```diff
@@ -16,85 +19,132 @@ export const MainView = () => {
+                        element={
+                            
+                                user ?  <Navigate to="/" /> : (
+                                    <Col md={5}>
+                                        <SignupView />
+                                    </Col>
+                                )
+                        }
+                    />
+                    <Route
+                        path="/profile"
+                        element={
+                            user ? ( 
+                            <Row className="justify-content-center">
+                                <Col sm={12} md={9} lg={7} >
+                                    
+                                        <ProfileView
+                                            token={token}
+                                            user={user}
+                                            movies={movies}
+                                            onSubmit={ setUser}
+                                        />
+                                     
+                                </Col>
+                            </Row>
+                            ) : (
+                                <Navigate to="/login" />
+                            )
+                        }
+                    />
+                    <Route
+                        path="/login"
+                        element={
+                                user ? <Navigate to="/" /> : (
+                                    <Col md={5}>
+                                        <LoginView onLoggedIn={(user, token) => {
+                                            setUser(user);
+                                            setToken(token);
+                                        }} />
+                                    </Col>
+                                )
+                        }
+                    />
+                    <Route
+                        path="/movies/:movieId"
+                        element={
+                            
+                                !user ? <Navigate to="/login" replace /> :
+                                movies.length === 0 ? 
+                                    <Col>The list is empty!</Col> :
+                                    <Col md={8}>
+                                        <MovieView movies={movies} />
+                                    </Col> 
+                        }
+                    />
+                    <Route
+                        path="/"
+                        element={
+                            
+                                !user ? <Navigate to="/login" replace /> :
+                                 movies.length === 0 ? <Col>The list is empty!</Col> :
+                                 movies.map(movie => (
+                                            <Col className="mb-4" key={movie.id} md={3}>
+                                                <MovieCard 
```

Comment:
I tried adding a movie to my favourite, and it threw an error because the `MovieCard` is missing the `user` props passed in `main-view`.

![image](https://github.com/behailu-2023/myFlix-client/assets/37598493/506784d9-3a36-458d-99f3-f6908e5eebef)